peps/pep-0780.rst

```diff
@@ -0,0 +1,447 @@
+
+The Python standard library already has mechanisms to provide information about
+the ABI of the interpreter, like the :data:`!sys.abiflags` attribute and the
+:mod:`sysconfig` module, but these do not provide all the `ABI Features`_ that
+are relevant for environment markers.
+The new attribute :data:`!sys.abi_features` is a natural extension to the
+existing mechanisms, and it is a simple and intuitive way to provide this
+information, including the information from :data:`!sys.abiflags`.
+
+A Forward Looking View on Free-Threaded Python
+----------------------------------------------
+
+:pep:`703`, the accepted proposal for free threading, states that the
+rollout of free-threaded Python should be gradual, which has been clarified
+by the Python Steering Council in `the PEP 703 acceptance post`_ to mean a
+three stage process over multiple releases. It is therefore important to make
+sure that the mechanisms in this PEP are useable for Python interpreters where
+either free-threading or non-free-threading could be the default or the only
+option.
+
+At the time of writing, free-threaded Python is in Phase I: experimental phase.
+In this phase, there is an acute need for the proposed environment markers to
+help with the transition to free-threaded Python as package authors gradually
+add support.
+
+As the number of packages with support increases, and particularly during
+Phase II: Supported-but-not-default phase, we still anticipate a strong need
+for the environment markers to help with the transition.
+
+As free-threaded Python enters into Phase III: Default phase, the need for
+the environment markers will decrease, though at this point it is not clear
+that the GIL-enabled Python will be completely phased out (it may remain
+available as a non standard build option). If it persists, the inverse need for
+the ABI feature detection may arise.
+
+Indeed, in all three phases it may be necessary for package authors to choose
+specific versions of their dependencies based on the ABI features, with a shift
+from GIL-enabled as default to free-threading as default over time.
+
+The ABI features are designed with this in mind to guarantee usefulness and
+simplicity for the foreseeable future in a changing Python ecosystem.
+
+Specification
+=============
+
+The keywords "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**",
+"**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**",
+and "**OPTIONAL**"" in this document are to be interpreted as described in
+:rfc:`RFC 2119 <2119>`.
+
+Concepts
+--------
+
+ABI features are a clear description of certain properties of the Python
+interpreter. While some of these features can be queried already today, they
+are not easily nor uniformly accessible.
+
+Hence, the proposed feature MUST be made available via the Python standard
+library as ``sys.abi_features``, and as the new environment marker variable
+``sys_abi_features``.
```

Comment:
Editorial: I don't think this subsection adds much, the first paragraph is covered above (in *Rationale* & *Motivation*) and the second paragraph is covered below (in the remainder of *Specification*).